esp/services/ws_ecl/ws_ecl_service.cpp

```diff
@@ -2001,7 +2001,7 @@ int CWsEclBinding::submitWsEclWorkunit(IEspContext & context, WsEclWuInfo &wsinf
 
     Owned<ISpan> clientSpan;
     ISpan * activeSpan = context.queryActiveSpan();
-    clientSpan.setown(activeSpan->createClientSpan("wsecl/SubmitWorkunit"));
+    clientSpan.setown(activeSpan->createClientSpan("run_workunit"));
```

Comment:
This is where the idea of PRODUCER spans gets interesting.  I guess the producer span would be created here?  or would we do it in workunit.dll when a workunit is created?  Then even the most basic eclagent/roxie/hthor job would be represented by a CONSUMER span?  And thor jobs may create multiple consumer spans.

Reply:
I agree, this is where the producer span would be.  I'm not sure that I agree it makes sense to differentiate a consumer span from a server span though.  In general the service does not know whether the parent span is waiting for it or not.  It may make sense with an internal producer/consumer model.

Reply:
@ghalliday just wanted to make a note to retain this thought.

I think the reason the implication for a consumer span rather than a server span is that a server span should show up as a sequential within the span of its parent.

If a THOR job itermitantly / asynchronously works on a workunit the spans would not correlate to the original parent at all which isn't what I think is expected of server spans.

Web app example

WEB server span 
---------------------------------------------------------------------------------
ROXIE server span1        ROXIE server span2          ROXIE server span3
-------------------           -------------------------     ---------------------------    
                                           SOAPCALL span1 
                                           ------------------ 

THOR job example

JOB async kickoff PRODUCER span (finished before any work really begins)
--

                                       THOR graph CONSUMER1   THOR graph CONSUMER2
                                       ------------------------------  -------------------------------

Reply:
What I am not sure about in this model is that the consumer does not really know whether the caller is waiting for it or not.   Does a webservice know if the caller is waiting for it?  On reflection maybe it does.  What happens if a server span sends an acknowledgement to the client before it finishes processing?  Is it require to create an internal producer/consumer span if that happens?
In which case anything that pulls something off a queue should be a consumer span, but anything that is processing an http request is a server span.  Definitely something to flesh out in a further discussion.

Reply:
@ghalliday, regarding "Does a webservice know if the caller is waiting for it?", I would think so in the case of simple synchronous HTTP.  But what about the async or even asymmetric case? Such as where a web server does HTTP/2 Push, or Server-Sent Events? That seems to match the PRODUCER pattern.  
[ Sorry if this is not a use case/pattern currently being considered or discussed. But as a discussion of span naming conventions that will extend beyond HPCC, I though the question is relevant ]

Reply:
WsECL specifically lets the caller decide if they want to have it return immediately giving them the WUID so they can retrieve the result, or will wait for the workunit to complete.

I'd consider:
1. having a flag in the workunit that determines whether graph processors should create server or consumer spans.
2. We aren't talking about HPCC behavior, rather about how this is recorded on the backend for external observability stacks to display.  In some sense we are choosing a diagram we want used? or perhaps tools will provide different functionality.  But in any case the reality of how those tools operate on the data will be the long term deciding factor for which representation is best.  Since they probably modeled current PRODUCER behavior around message queues and databases we could see how well the representation of those in grafana and elastic match with async workunits.